src/ServiceControl.Audit.AcceptanceTests/Auditing/When_running_setup_command__integration_test.cs

```diff
@@ -0,0 +1,300 @@
+            Assert.IsTrue(indexesFileContent.Any(i => i.Contains("MessagesViewIndex")), "RavenDB database does not contain expected index MessagesViewIndex");
+        }
+
+        #region Setup
+        [SetUp]
+        public void Setup()
+        {
+            var transportIntegration = (ITransportIntegration)TestSuiteConstraints.Current.CreateTransportConfiguration();
+            Console.WriteLine($"\tConfigured transport configuration: {transportIntegration.Name}");
+            //Mark the test as inconclusive if when running in the CI SQL Server is not installed.
+#if !DEBUG
+            if (transportIntegration.Name.Equals("SQL Server", StringComparison.InvariantCultureIgnoreCase) == false)
+            {
+                _inconclusiveRun = true;
+                Assert.Inconclusive($"'");
+                return;
+            }
+#endif
+            DeleteEventViewerEventSource();
+            SetupWorkingDirectory();
+            _sqlLocalDbInstance = SqlLocalDb.CreateNewIn(_workingDirectory);
+
+        }
+
+        static void DeleteEventViewerEventSource()
+        {
+            try
+            {
+                if (EventLog.SourceExists(CreateEventSource.SourceName))
+                {
+                    EventLog.DeleteEventSource(CreateEventSource.SourceName);
+                }
+            }
+            catch
+            {
+                // ignored
+            }
+
+            Assert.IsTrue(EventLog.SourceExists(CreateEventSource.SourceName) == false,
+                $"Running this test requires elevated privileges. The test setup couldn't confirm windows event log source {CreateEventSource.SourceName} doesn't exists before starting the test execution");
+        }
+
+        void SetupWorkingDirectory()
+        {
+            _workingDirectory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+            Directory.CreateDirectory(_workingDirectory);
+        }
+
+        #endregion Setup
+
+        #region TearDown
+        [TearDown]
+        public void TearDown()
+        {
+            if (_inconclusiveRun)
+            {
+                return;
+            }
+
+            _sqlLocalDbInstance?.Detach();
+            DeleteFolder(_workingDirectory);
+        }
+        static void DeleteFolder(string path)
+        {
+            DirectoryInfo emptyTempDirectory = null;
+
+            if (!Directory.Exists(path))
+            {
+                return;
+            }
+
+            try
+            {
+                emptyTempDirectory = new DirectoryInfo(Path.Combine(Path.GetTempPath(), Path.GetRandomFileName()));
+                emptyTempDirectory.Create();
+                var arguments = $"\"{emptyTempDirectory.FullName}\" \"{path.TrimEnd('\\')}\" /W:1  /R:1 /FFT /MIR /NFL";
+                using (var process = Process.Start(new ProcessStartInfo("robocopy")
```

Comment:
what is this required for?

Reply:
At first, I tried to just use Directory.Delete but it failed. I copied this code over from the Acceptance Tests base classes that actually also rely on invoking the SetupBootstraper